lib/solidus_subscriptions/order_creator.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module SolidusSubscriptions
+  class OrderCreator
+    def initialize(installment)
+      @installment = installment
+    end
+
+    def call
+      ::Spree::Order.create(
+        user: installment.subscription.user,
+        email: installment.subscription.user.email,
+        store: installment.subscription.store || ::Spree::Store.default,
+        subscription_order: true,
+        subscription: installment.subscription,
+        currency: installment.subscription.currency,
+        **extra_attributes
+      )
+    end
+
+    protected
```

Comment:
Should this be [private rather than protected](https://stackoverflow.com/questions/3534449/why-does-ruby-have-both-private-and-protected-methods)?

Reply:
Yeah, it could be. I can see and advantage to keeping it protected (a subclass of OrderCreator implementing a decorator might want to call it with a `decorated.extra_attributes` it instead of overriding the method) but at this point this is a lot of overengineering 😅